src/main/java/sx/blah/discord/api/internal/DiscordClientImpl.java

```diff
@@ -178,8 +178,8 @@ private DiscordClientImpl(long timeoutTime, int maxMissedPingCount, boolean isDa
 		this.isBot = isBot;
 		this.reconnectAttempts = reconnectAttempts;
 		this.dispatcher = new EventDispatcher(this);
-		this.loader = new ModuleLoader(this);
 		this.shardCount = shardCount;
+		this.loader = new ModuleLoader(this);
```

Comment:
This is on purpose
The constructor for ModuleLoader calls isReady() which accesses shardCount which causes NPEs to rain down like its the apocalypse